### PR TITLE
Always print addErrorContext

### PR DESCRIPTION
**Pull request by roberth**
# Motivation

`addErrorContext` traces are sufficiently valuable and usually terse, so they should always be shown.

# Context

- Closes https://github.com/NixOS/nix/issues/7553
  - no `--no-trace`, which might actually be confusing, because of `--no-show-trace`, which is a simple negation of `--show-trace`.

# Priorities and Process

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

### CODE REVIEW

File: tests/functional/lang/eval-fail-addErrorContext-example.flags [hunk 1]
```diff
@@ -0,0 +1 @@
+--eval --strict --no-show-trace
```
Comment (9999years) [inline tests/functional/lang/eval-fail-addErrorContext-example.flags:1]: Missing EOL at end of file https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Comment (roberth) [reply tests/functional/lang/eval-fail-addErrorContext-example.flags]: Wasn't meant to be a line, but bash handles that fine.

File: src/libutil/error.hh [hunk 2]
```diff
@@ -61,9 +61,16 @@ void printCodeLines(std::ostream & out,
     const Pos & errPos,
     const LinesOfCode & loc);
 
+enum struct TraceKind {
+    Other,
+    /** Produced by builtins.addErrorContext. Always printed. */
+    Custom,
+};
```
[the remark this answers is not in the document]
Comment (roberth) [reply src/libutil/error.hh]: Applied. Thanks!

**Review by 9999years — APPROVED** [review]

**Comment by 9999years** [comment]
Looks good. These still need work. (I can imagine it being frustrating that source code positions are not shown for these messages.)

**Pull request merged**